### PR TITLE
Docs: Reorganise Adapters docs, move higher in sidebar due to importance, add nextjs page

### PR DESCRIPTION
**Pull request by Nick-Lucas**
This work creates a more cohesive location for Adapters documentation, and normalises our terminology.

* We mostly refer to Adapters as Adapters, but "API Handler" is seemingly used for this one api-handlers doc. It has been renamed to be the introduction to the Adapters tree
* The existing api-handlers doc has been stripped down massively, since it largely covered content which can be more conveniently covered by individual adapters, and in a nextjs adapter guide - an adapter should be easy to choose, so users can quickly go the right place
* We now recommend the standalone adapter in some scenarios, this is an intentional push for less experienced users to default to it and lower the barrier to entry
* NextJS has its own integration guide, but not an adapters page, this was a small gap in the flow for NextJS users, which has been plugged with the content lifted from the old api-handlers page. We also link back to this page now from the Nextjs setup guide (might want to do something more tightly integrated in the future)

image

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?

## ✅ Checklist

- [ ] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: www/vercel.json [hunk 1]
```diff
@@ -6,8 +6,8 @@
       "permanent": true
     },
     {
-      "source": "/docs/aborting-procedure-calls",
```
Comment (Nick-Lucas) [inline www/vercel.json:9]: (This was a duplicate entry)

File: www/docs/server/adapters/nextjs.md [hunk 2]
```diff
@@ -1,37 +1,57 @@
...
+          <li><a href="https://github.com/trpc/trpc/tree/main/examples/next-minimal-starter">Source</a></li>
+        </ul>
+      </td>
+    </tr>
+  </tbody>
+</table>
 
 ## Next.js example
 
+Hosting tRPC within a Next.js server is straight-forward, just create an API Handler like `pages/api/trpc/[trpc].ts` and export a created handler like below.
```
Comment (juliusmarminge) [inline www/docs/server/adapters/nextjs.md:36]:
```suggestion
Serving your tRPC router in a Next.js project is straight-forward. Just create an API handler in `pages/api/trpc/[trpc].ts` as shown below:
```

File: www/docs/server/routers.md [hunk 3]
```diff
@@ -31,7 +31,7 @@ You'll notice we are exporting certain methods of the `t` variable here rather t
 
 Next, let's define a router with a procedure to use in our application. We have now created an API "endpoint".
 
-In order for these endpoints to be exposed to the frontend, the outermost router (in the example below, this is the `appRouter`) needs to be passed to the [API Handler](/docs/server/api-handler)'s `router` argument.
+In order for these endpoints to be exposed to the frontend, your [Adapter](/docs/server/adapters) should be configured with `appRouter` instance.
```
Comment (juliusmarminge) [inline www/docs/server/routers.md:34]:
```suggestion
In order for these endpoints to be exposed to the frontend, your [Adapter](/docs/server/adapters) should be configured with your `appRouter` instance.
```

File: www/docs/server/adapters/nextjs.md (hunk 2 above)
Comment (Nick-Lucas) [reply www/docs/server/adapters/nextjs.md:36]: Much improved thanks <3

File: www/docs/server/adapters/standalone.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ---
 id: standalone
-title: Standalone Usage
+title: Standalone Adapter
 sidebar_label: Standalone
 slug: /server/adapters/standalone
 ---
```
[the remark this answers is not in the document]
Comment (Nick-Lucas) [reply www/docs/server/adapters/standalone.md:6]: Great idea, I'll do this tonight!

File: www/docs/server/adapters-intro.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+slug: /server/adapters
+---
+
+tRPC is not a backend on its own, but is instead mounted inside of other hosts, such as a simple [Node.js HTTP Server](adapters/standalone), [Express](adapters/express), or even [Next.js](adapters/nextjs). Most tRPC features are the same no matter which backend you choose. **Adapters** act as the glue between the host system and your tRPC API.
+
+Adapters typically follow some common conventions, allowing you to set up context creation via `createContext`, and globally handle errors via `onError`, but importantly allow you to choose an appropriate host for your application.
+
+We support many modes of hosting an API, which you will find documented here.
+
+:::tip
```
[the remark this answers is not in the document]
Comment (Nick-Lucas) [reply www/docs/server/adapters-intro.md:14]: See what you think of this, I went a bit further but I think a nice summary was what this lacked

File: www/docs/server/adapters-intro.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+title: Adapters
+sidebar_label: Adapters
+slug: /server/adapters
+---
+
+tRPC is not a backend on its own, but is instead mounted inside of other hosts, such as a simple [Node.js HTTP Server](adapters/standalone), [Express](adapters/express), or even [Next.js](adapters/nextjs). Most tRPC features are the same no matter which backend you choose. **Adapters** act as the glue between the host system and your tRPC API.
+
+Adapters typically follow some common conventions, allowing you to set up context creation via `createContext`, and globally handle errors via `onError`, but importantly allow you to choose an appropriate host for your application.
+
+We support many modes of hosting an API, which you will find documented here. You might have a server-based API, and want the [Standalone](adapters/standalone), [Express](adapters/express), or [Fastify](adapters/fastify) adapters. You might want a serverless solution and choose [AWS Lambda](aws-lambda) or [Fetch / Edge](fetch). You might have a full-stack framework and want a full integration like [Next.js](adapters/nextjs), or you could use the [Fetch / Edge](fetch) adapter with Next.js, Astro, Remix, or Solidstart.
```
Comment (juliusmarminge) [inline www/docs/server/adapters-intro.md:12]:
```suggestion

We support many modes of hosting an API, which you will find documented here. You might have a server-based API, and want the [Standalone](adapters/standalone), [Express](adapters/express), or [Fastify](adapters/fastify) adapters. You might want a serverless solution and choose [AWS Lambda](adapters/aws-lambda) or [Fetch / Edge](adapters/fetch). You might have a full-stack framework and want a full integration like [Next.js](adapters/nextjs), or you could use the [Fetch / Edge](adapters/fetch) adapter with Next.js, Astro, Remix, or Solidstart.
```

File: www/docs/server/adapters-intro.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+Adapters typically follow some common conventions, allowing you to set up context creation via `createContext`, and globally handle errors via `onError`, but importantly allow you to choose an appropriate host for your application.
+
+We support many modes of hosting an API, which you will find documented here.
+
+- For serverful APIs, you might want our [Standalone](adapters/standalone) adapter, or use the [Express](adapters/express) or [Fastify](adapters/fastify) adapters to hook into your existing APIs.
+- You might want a serverless solution and choose [AWS Lambda](adapters/aws-lambda) or [Fetch](adapters/fetch). You might have a full-stack framework and want a full integration like [Next.js](adapters/nextjs), or you could use the [Fetch](adapters/fetch) adapter with Next.js, Astro, Remix, or SolidStart.
+- For edge runtimes, refer to the [Fetch](adapters/fetch) adapter that builds on Web APIs.
```
Comment (Nick-Lucas) [inline www/docs/server/adapters-intro.md:16]:
```suggestion
- For serverful APIs, you might want our [Standalone](adapters/standalone) adapter, or use the [Express](adapters/express) or [Fastify](adapters/fastify) adapters to hook into your existing APIs
- You might want a serverless solution and choose [AWS Lambda](adapters/aws-lambda) or [Fetch](adapters/fetch). You might have a full-stack framework and want a full integration like [Next.js](adapters/nextjs), or you could use the [Fetch](adapters/fetch) adapter with Next.js, Astro, Remix, or SolidStart
- For edge runtimes, refer to the [Fetch](adapters/fetch) adapter that builds on Web APIs.
```

I've been taught that lists should have a `.` on the final item only, good shout to turn it into a list though :)

File: www/docs/server/adapters/nextjs.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+    res.writeHead(200);
+    return res.end();
+  }
+
+  // finally pass the request on to the tRPC handler
+  return nextApiHandler(req, res);
+}
+```
+
+## Route Handlers
```
[the remark this answers is not in the document]
Comment (Nick-Lucas) [reply www/docs/server/adapters/nextjs.md:99]: LGTM!! Thanks for your changes, I made one tweak but it's all very welcome polish 🙂

**Review by juliusmarminge — APPROVED** [review]

**Pull request merged**